www/js/survey/input-matcher.js

```diff
@@ -116,5 +116,37 @@ angular.module('emission.survey.inputmatcher', ['emission.plugin.logger'])
     Logger.log("Returning mostRecentEntry "+printUserInput(mostRecentEntry));
     return mostRecentEntry;
   }
+
+  // return array of matching trip additions
+  im.getTripAdditionsForTrip = function(trip, tripAdditionList) {
+    if (tripAdditionList === undefined) {
+      Logger.log("In getTripAdditionsForTrip, no trip addition input, returning []");
+      return undefined;
+    }
+
+    // remove deleted additions
+    tripAdditionList.forEach(addition => {
+        if (addition.data.status == "DELETED") {
+            // find and remove the original addition
+            const i = tripAdditionList.findIndex(a =>
+                a.data.end_ts == addition.data.end_ts
+                && a.data.start_ts == addition.data.start_ts);
+            tripAdditionList.splice(i, 1)
+            // also remove this deletion signifier
+            tripAdditionList.splice(tripAdditionList.indexOf(addition), 1);
+        }
+    });
+
+    // filter out trip additions that do not start withing the bounds of the trip
+    const matchingAdditions = tripAdditionList.filter((additionCandidate) =>
+        additionCandidate.data.start_ts >= trip.start_ts &&
+        additionCandidate.data.start_ts <= trip.end_ts);
```

Comment:
Perhaps naively, I thought we could avoid the complexity of the other matching function. Maybe that is because I don't fully understand how drafts work. Why would an addition's time bounds not fall within the trip's time bounds?

Reply:
one of the issues we had reported by people earlier is that unprocessed trips wouldn't show up in the app. So if you took a trip, at the end of the trip, you would still see a "blank diary". So people didn't know if the app was working.

to address this, we display draft aka "green background" entries for unprocessed trips which we kind of hack together by looking at the transitions of the trip start and trip end from the data collection FSM. These are currently displayed in the diary, but not in the label screen.

so a concrete example would be:
- I just got home, 5 mins later, the FSM in the app has detected that the trip ended, and has attempted to push the trip data to the server
   - if it was unable to push (due to lack of network), the trip would be draft
   - if it was pushed, but the pipeline has not yet run, the trip would be draft

so note that the FSM on the phone will always have a bit of a lag for both trip start and trip end.
- On trip start, we need to exit the geofence or be moving for some substantive time before we start tracking, otherwise, we will start tracking everytime somebody goes to the kitchen of their house.
- On trip end, we need to wait until the user has been stationary for a while before we detect a trip end and stop tracking, otherwise we will stop the trip at every red light.

So the transitions from the phone FSM will lag the actual start and end in general.

In the processing, as part of the trip segmentation and clean and resample stages we attempt to fill these gaps. 

- Because we are post-processing, we know the end place and we can work backwards and find when we entered the end place and pick that as the trip end.
- Similarly, at the trip start, we know the last end place, so at trip start, we can "join" back to the previous place and extrapolate the time required to do so, and pick that as trip start.